client/frontend/src/SearchResult/SearchResult.js

```diff
@@ -53,6 +64,7 @@ function SearchResult(props) {
     fetchDocuments(docUrls).then(() => {
       changeSearchTime((Date.now() - startTime) / 1000);
       changeLoading(false);
+      stillLoading = false;
```

Comment:
Can you add a check from the first data to see whether or not success was 1? If not's 1 then there was an error connecting to ranker. If its not 1, automatically trigger the timeout redirect